### ISSUE TITLE
Community Fork?

### ISSUE THREAD

**Issue by roock** [infinityworksltd/hpilo-exporter#21]
I see a lot of people have currently forked this repository working independently on new feature and keeping it maintained.  What do you guy think of joining forces together and drive this as a community effort?

Tagging everyone how contributed on a fork within the last year:

@cbw @mhollstein @srkaviani @longpaul @n27051538 @baonq-me @root-ali  @wardellc 

Please let me know what you think and/or willing to do this as a community.

**Comment by srkaviani** [comment]
Very good.
I agree to join this community.

**Comment by baonq-me** [comment]
What do you suggest?

**Comment by roock** [comment]
I have created the [hpilo-exporter github organization](https://github.com/hpilo-exporter) and sent you invites. I would suggest that we continue the development of this project there and start further discussions  [there](https://github.com/hpilo-exporter/internal/discussions).

**Comment by cbw** [comment]
I'm open to participating. Thanks @roock for driving this!

**Comment by baonq-me** [comment]
> I have created the [hpilo-exporter github organization](https://github.com/hpilo-exporter) and sent you invites. I would suggest that we continue the development of this project there and start further discussions [there](https://github.com/hpilo-exporter/internal/discussions).

Can you be more specific, like which feature or problem should be addressed?

**Comment by longpaul** [comment]
Sadly the exporter is not actively maintained. That's why I'm also willing to help

**Comment by roock** [comment]
> Can you be more specific, like which feature or problem should be addressed?

The problem I would like to solve is that currently there are 10+ forks of the original repository, all with different features and/or features which have been implemented multiple times independently. I think everyone would benefit if we have one repository which contains all features and is kept uptodate.

I can understand that you have already spend a lot of time to merge some of the changes together, but your repository is lacking some important things like python3 support or automated building of docker image.

**Comment by n27051538** [comment]
Hello, dears! 

I have added this to exporter a few years ago:
 - support for Python 3.6.  
 - ilo_user, ilo_password, ilo_port may be preset via environment.  
 - storage health information from iLO (cache, controller, logical drives, physical drives).  
 -  temperature values information from iLO.
 - per-fan and per-power-supply statuses.
 - OA info for Blade servers
 - Server ON status.
 - Alert rules for Prom
 - Also I have Grafana Dashboard about, but it is not published to github. 

Since this modification it works well on all HPE servers that I have to monitor.
I am not using docker still, so DockerFile is without any mod and not tested.

**Comment by baonq-me** [comment]
Here is my dashboard (of course, it works with my fork https://github.com/baonq-me/hpilo-exporter)
https://gist.github.com/baonq-me/75f3041fea0e01f74d6a170c53c95a5e

![image](https://user-images.githubusercontent.com/1851037/231943551-cac954a9-fc5d-4319-8f5a-a3f56e610378.png)

**Comment by roock** [comment]
> Here is my dashboard (of course, it works with my fork https://github.com/baonq-me/hpilo-exporter) https://gist.github.com/baonq-me/75f3041fea0e01f74d6a170c53c95a5e

Yeah, your Grafana Dashboard looks great, I'm already using it 😄  As you just stated, it works with YOUR fork, I think everyone would benefit from one shared repository where all the improvements are collected and kept uptodate. And which is easy to find and you don't have to look in 5 different forks to get all the features.

Also your Grafana Dashboard contains some hardcoded references (e.g. to datasources and hostnames).

**Comment by roock** [comment]
> Hello, dears!
> 
> I have added this to exporter a few years ago:
> 
> * support for Python 3.6.
> * ilo_user, ilo_password, ilo_port may be preset via environment.
> * storage health information from iLO (cache, controller, logical drives, physical drives).
> * temperature values information from iLO.
> * per-fan and per-power-supply statuses.
> * OA info for Blade servers
> * Server ON status.
> * Alert rules for Prom
> * Also I have Grafana Dashboard about, but it is not published to github.
> 
> Since this modification it works well on all HPE servers that I have to monitor. I am not using docker still, so DockerFile is without any mod and not tested.

Yeah, you did a great job with the improvements, I would really like to see them in a shared repository. (Unfortunately it's a bit hard to cherry-pick your changes, since you make more or less complete rewrite 😄

**Comment by srkaviani** [comment]
> Here is my dashboard (of course, it works with my fork https://github.com/baonq-me/hpilo-exporter) https://gist.github.com/baonq-me/75f3041fea0e01f74d6a170c53c95a5e
> 
> ![image](https://user-images.githubusercontent.com/1851037/231943551-cac954a9-fc5d-4319-8f5a-a3f56e610378.png)

Your Grafana Dashboard is very good. but it contains some hardcoded references. for fix this issue please export dashboard with option "Export for sharing externally".
and I suggest to share this dashboard and other future dashboard of this exporter to grafana.com (same as : https://grafana.com/orgs/soroush7339)

**Comment by baonq-me** [comment]
> > Here is my dashboard (of course, it works with my fork https://github.com/baonq-me/hpilo-exporter) https://gist.github.com/baonq-me/75f3041fea0e01f74d6a170c53c95a5e
> > ![image](https://user-images.githubusercontent.com/1851037/231943551-cac954a9-fc5d-4319-8f5a-a3f56e610378.png)
> 
> Your Grafana Dashboard is very good. but it contains some hardcoded references. for fix this issue please export dashboard with option "Export for sharing externally". and I suggest to share this dashboard and other future dashboard of this exporter to grafana.com (same as : https://grafana.com/orgs/soroush7339)

Thanks for your idea. I'll bring the dashboard to grafana.com.

**Comment by roock** [comment]
I've started creating a repository with all the combined changes of the different forks including a matching Grafana Dashboard and a working Dockerfile with Python3.
https://github.com/hpilo-exporter/hpilo-exporter

**Comment by mhollstein** [comment]
Hello Roman,You can use everything from my forkand put it into your project if you want. Best Regards,MichaelGesendet mit der mobilen Mail AppAm 06.04.23 um 17:00 schrieb Roman Pertl
                
            
            
                
                    Von: "Roman Pertl" ***@***.***>Datum: 6. April 2023An: "infinityworks/hpilo-exporter" ***@***.***>Cc: "mhollstein" ***@***.***>,"Mention" ***@***.***>Betreff: [infinityworks/hpilo-exporter] Community Fork? (Issue #21)
                
                
I see a lot of people have currently forked this repository working independently on new feature and keeping it maintained.  What do you guy think of joining forces together and drive this as a community effort?
Tagging everyone how contributed on a fork within the last year:
@cbw @mhollstein @srkaviani @longpaul @n27051538 @baonq-me @root-ali  @wardellc
Please let me know what you think and/or willing to do this as a community.

**Comment by jhonyh** [comment]
Hi guys,

I am interested to have this exporter, but please help me how to install and make it work properly.

I had installed and run the dcoker as follows:
docker run -p 9416:9416 hpilo-exporter:latest
with result:
CONTAINER ID   IMAGE                     COMMAND                  CREATED        STATUS        PORTS                                       NAMES
8e2472b7bc2d   hpilo-exporter:latest     "hpilo-exporter"         14 hours ago   Up 14 hours   0.0.0.0:9416->9416/tcp, :::9416->9416/tcp   jovial_archimedes

Starting exporter on: http://0.0.0.0:9416/metrics
Press Ctrl+C to quit

but when I tried to curl it:
[root@grafaprome ~]# curl -v 'http://10.212.70.200:9416/metrics?ilo_host=10.117.3.111&ilo_port=443&ilo_user=administrator&ilo_password=P@ssw0rd!'
* About to connect() to 10.212.70.200 port 9416 (#0)
*   Trying 10.212.70.200...
* Connected to 10.212.70.200 (10.212.70.200) port 9416 (#0)
> GET /metrics?ilo_host=10.117.3.111&ilo_port=443&ilo_user=administrator&ilo_password=P@ssw0rd! HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 10.212.70.200:9416
> Accept: */*
>
* Empty reply from server
* Connection #0 to host 10.212.70.200 left intact
curl: (52) Empty reply from server
[root@grafaprome ~]#

the prometheus yml is as follows:
 - job_name: 'hpilo-agent'
    scrape_interval: 1m
    params:
    ilo_port: ['443']
    ilo_user: ['administrator']
    ilo_password: ['P@ssw0rd!']
    static_configs:
     - targets:
       - '10.117.3.111
#    - 10.117.3.116
#    - 10.117.3.117
#    - 10.117.3.118
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_ilo_host
    - source_labels: [__param_ilo_host]
      target_label: ilo_host
    - target_label: __address__
      #replacement: 10.212.70.200:8082  # hpilo exporter
      replacement: 10.212.70.200:9416  # hpilo exporter

Did I any missed in configuration?
Please help to make it done well to show it on the dashboard.

Thanks in advance

**Comment by srkaviani** [comment]
Hi

> Hi guys,
> 
> I am interested to have this exporter, but please help me how to install and make it work properly.
> 
> I had installed and run the dcoker as follows: docker run -p 9416:9416 hpilo-exporter:latest with result: CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES 8e2472b7bc2d hpilo-exporter:latest "hpilo-exporter" 14 hours ago Up 14 hours 0.0.0.0:9416->9416/tcp, :::9416->9416/tcp jovial_archimedes
> 
> Starting exporter on: http://0.0.0.0:9416/metrics Press Ctrl+C to quit
> 
> but when I tried to curl it: [root@grafaprome ~]# curl -v 'http://10.212.70.200:9416/metrics?ilo_host=10.117.3.111&ilo_port=443&ilo_user=administrator&ilo_password=P@ssw0rd!'
> 
> * About to connect() to 10.212.70.200 port 9416 (#0)
> * Trying 10.212.70.200...
> * Connected to 10.212.70.200 (10.212.70.200) port 9416 (#0)
> 
> > GET /metrics?ilo_host=10.117.3.111&ilo_port=443&ilo_user=administrator&ilo_password=P@ssw0rd! HTTP/1.1
> > User-Agent: curl/7.29.0
> > Host: 10.212.70.200:9416
> > Accept: _/_
> 
> * Empty reply from server
> * Connection #0 to host 10.212.70.200 left intact
>   curl: (52) Empty reply from server
>   [root@grafaprome ~]#
> 
> the prometheus yml is as follows:
> 
> * job_name: 'hpilo-agent'
>   scrape_interval: 1m
>   params:
>   ilo_port: ['443']
>   ilo_user: ['administrator']
>   ilo_password: ['P@ssw0rd!']
>   static_configs:
>   
>   * targets:
>     
>     * '10.117.3.111
> 
> # - 10.117.3.116
> # - 10.117.3.117
> # - 10.117.3.118
> ```
> relabel_configs:
> - source_labels: [__address__]
>   target_label: __param_ilo_host
> - source_labels: [__param_ilo_host]
>   target_label: ilo_host
> - target_label: __address__
>   #replacement: 10.212.70.200:8082  # hpilo exporter
>   replacement: 10.212.70.200:9416  # hpilo exporter
> ```
> 
> Did I any missed in configuration? Please help to make it done well to show it on the dashboard.
> 
> Thanks in advance

Hi,
this repository is not maintained .
Please use this and if you have any issues you can create issue there:
https://github.com/hpilo-exporter/hpilo-exporter

**Comment by jhonyh** [comment]
Hi Guys,

How can I start this hpilo-exporter to obtain multiple hosts? I also want this service shall be started up after once the server reboot. 
I was done with export the environment like ILO_HOST=ip target, ILO_PORT=443, ILO_USER=username, ILO_PASSWORD=userpassword then execute hpilo-exporter --address=ip exporter --port=9416 --endpoint="/metrics". But it is only server a single target host, I want can get multiple target host, how can we do it?

Thanks in advanced

**Comment by jhonyh** [comment]
the dashboard is now does not show the metric, why? how long it will scrap back?
